mod-source-record-manager-server/src/main/java/org/folio/services/RecordsPublishingServiceImpl.java

```diff
@@ -165,35 +158,21 @@ private boolean isRecordReadyToSend(Record record) {
     return true;
   }
 
-  /**
-   * Provides external parameters for the MARC-to-Instance mapping process
-   *
-   * @param snapshotId  - snapshotId
-   * @param okapiParams okapi connection parameters
-   * @return mapping parameters
-   */
-  private Future<MappingParameters> getMappingParameters(String snapshotId, OkapiConnectionParams okapiParams) {
-    return mappingParametersProvider.get(snapshotId, okapiParams);
-  }
-
   /**
    * Prepares eventPayload with record and profileSnapshotWrapper
    *
    * @param record                 record to send
    * @param profileSnapshotWrapper profileSnapshotWrapper to send
-   * @param mappingRules           rules for default instance mapping
-   * @param mappingParameters      mapping parameters
    * @param params                 connection parameters
    * @return dataImportEventPayload
    */
   private DataImportEventPayload prepareEventPayload(Record record, ProfileSnapshotWrapper profileSnapshotWrapper,
-                                                     JsonObject mappingRules, MappingParameters mappingParameters, OkapiConnectionParams params,
-                                                     String eventType) {
-    HashMap<String, String> context = payloadContextBuilder.buildFrom(record, mappingRules, mappingParameters);
+                                                     OkapiConnectionParams params, String eventType) {
+    HashMap<String, String> context = payloadContextBuilder.buildFrom(record,
+      Objects.isNull(profileSnapshotWrapper) ? StringUtils.EMPTY : profileSnapshotWrapper.getId());
```

Comment:
As alternative we can use: `Optional.ofNullable(profileSnapshotWrapper).map(ProfileSnapshotWrapper ::getId).orElse(StringUtils.EMPTY)`.
By the way, maybe better to use null instead of empty string in this case

Reply:
At the stage, the variable should not be null, so the check is moved. Thank you.